### PR TITLE
23.07 doc fixes

### PR DESCRIPTION
**Pull request by dagardner-nv**
## Description
* Update minimum driver version to match rapids 23.06
* Update example Kafka output
* Update docstring for `DeserializeStage` such that it reads better on the CLI
* Prevent property methods from being displayed side-by-side
* Fix broken code snippet in `docs/source/developer_guide/guides/3_simple_cpp_stage.md`, create a unittest mimicking the snippet to prevent it from going out of date again.
* Fix doscstrings to match function signatures
* Misc other fixes

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by dagardner-nv** [comment]
prop_methods
Example of tiled property methods

**Review by drobison00 — APPROVED** [review]

**Review by mdemoret-nv — APPROVED** [review]
Looks good. Thanks for the update.

File: docs/source/developer_guide/guides/2_real_world_phishing.md [hunk 1]
```diff
@@ -435,12 +435,13 @@ pipeline.add_stage(SerializeStage(config))
...
 
-Note that we didn't specify the output format. In our example, the result file contains the extension `.jsonlines`. Morpheus will infer the output format based on the extension. At time of writing the extensions that Morpheus will infer are: `.csv`, `.json` & `.jsonlines`
+Note that we didn't specify the output format. In our example, the result file contains the extension `.jsonlines`. Morpheus will infer the output format based on the extension. At time of writing the extensions that Morpheus will infer are: `.csv`, `.json`, `.jsonlines` & `.parquet`
 
 To explicitly set the output format we could specify the `file_type` argument to the `WriteToFileStage` which is an enumeration defined in `morpheus.common.FileTypes`. Current values defined are:
 * `FileTypes.Auto`
-* `FileTypes.JSON`
 * `FileTypes.CSV`
+* `FileTypes.JSON`
+* `FileTypes.PARQUET`
```
Comment (efajardo-nv) [inline docs/source/developer_guide/guides/2_real_world_phishing.md:444]: `WriteToFileStage` does not currently support `FileTypes.PARQUET`.

**Review by efajardo-nv — APPROVED** [review]

**Pull request merged**